Solutions/GoogleWorkspaceReports/Hunting Queries/GWorkspaceDocumentCopiedToPrivateDrive.yaml

```diff
@@ -0,0 +1,34 @@
+id: 69e8a40f-6508-4f43-8eef-2f78ad6174df
+name: GWorkspace - Document Copied from Share Drive to Private Drive
+description: |
+  This hunting query searches for document copy activity from shared drive to a private drive, potential sign of data exfiltration.
+  https://www.mitiga.io/blog/mitiga-security-advisory-lack-of-forensic-visibility-with-the-basic-license-in-google-drive
+severity: Medium
+requiredDataConnectors:
+  - connectorId: GoogleWorkspaceReportsAPI
+    dataTypes:
+      - GWorkspaceActivityReports
+tactics:
+  - Exfiltration
+  - Impact
+relevantTechniques:
+  - T1537
+  - T1565
+query: |
+  GWorkspaceActivityReports
+  | where TimeGenerated > ago(24h)
+  | where EventMessage =~ 'source_copy'
+  | project TimeGenerated, ActorEmail, SrcIpAddr, DocTitle
+  | extend Name = tostring(split(ActorEmail,'@',0)[0]), UPNSuffix = tostring(split(ActorEmail,'@',1)[0])
```

Comment:
We would need entity mapping of the kind "| extend Entity_0_Identifier = value"

Reply:
Added done.